processor/tailsamplingprocessor/processor.go

```diff
@@ -365,7 +382,14 @@ func (tsp *tailSamplingSpanProcessor) processTraces(resourceSpans ptrace.Resourc
 			traceTd := ptrace.NewTraces()
 			appendToTraces(traceTd, resourceSpans, spans)
 			tsp.releaseSampledTrace(tsp.ctx, id, traceTd)
-			tsp.telemetry.ProcessorTailSamplingEarlyReleasesFromCacheDecision.Add(tsp.ctx, int64(len(spans)))
+			tsp.telemetry.ProcessorTailSamplingEarlyReleasesFromCacheDecision.
+				Add(tsp.ctx, int64(len(spans)), metric.WithAttributes(attribute.String("decision", "sample")))
```

Comment:
Interstingly, metric.WithAttributes performs very poorly, so it should be avoided on the hot path. I typically have a static var for that, and refer it later. 

Using metric.WithAttributeSet should not cause any allocations on the hot path.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b95672e0a229b658baca5c6cfc557dba1ad90b34/exporter/loadbalancingexporter/wrapped_exporter.go#L36